java-frontend/src/main/java/org/sonar/java/se/ExplodedGraph.java

```diff
@@ -126,13 +129,23 @@ public void setParent(@Nullable Node parent) {
             learnedConstraints.add(new LearnedConstraint(sv, c));
           }
         });
+        programState.values.forEach((s, sv) -> {
+         if(parent.programState.getValue(s)!= sv) {
+           learnedSymbols.add(new LearnedValue(sv, s));
+         }
+        });
       }
     }
 
-    public List<LearnedConstraint> learnedConstraints() {
+    public List<LearnedConstraint> getLearnedConstraints() {
```

Comment:
if you want to use that getter, then you may want to change the visibility of the associated field, which is currently also `public final`.